### PR TITLE
[APB-5124][BS] Updated Relationship query endpoint to accept reference Number

### PR DESCRIPTION
**Pull request by barrysims**
I wasn't sure of the best way of approaching this, so I went for the smallest change.

I'm a bit uneasy about jamming this into DesStubController if this is really stubbing the IF service.  Any thoughts?

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -51,7 +51,7 @@ POST          /tax-enrolments/users/:userId/enrolments/:enrolmentKey
...
 GET           /registration/relationship/nino/:nino                                                  @uk.gov.hmrc.agentsexternalstubs.controllers.DesStubController.getLegacyRelationshipsByNino(nino: String)
-GET           /registration/relationship                                                             @uk.gov.hmrc.agentsexternalstubs.controllers.DesStubController.getRelationship(idtype: Option[String] ?= None, `ref-no`: Option[String] ?= None, arn: Option[String] ?= None, agent: Boolean, `active-only`: Boolean, regime: String, from: Option[String] ?= None, to: Option[String] ?= None, relationship: Option[String] ?= None, `auth-profile`: Option[String] ?= None)
+GET           /registration/relationship                                                             @uk.gov.hmrc.agentsexternalstubs.controllers.DesStubController.getRelationship(idtype: Option[String] ?= None, `ref-no`: Option[String] ?= None, `referenceNumber`: Option[String] ?= None, arn: Option[String] ?= None, agent: Boolean, `active-only`: Boolean, regime: String, from: Option[String] ?= None, to: Option[String] ?= None, relationship: Option[String] ?= None, `auth-profile`: Option[String] ?= None)
```
Comment (Leeroyc720) [inline conf/app.routes:54]: I'm not sure if we still need this other code `ref-no`: Option[String] ?= None, as we now call it referenceNumber?

Comment (barrysims) [reply conf/app.routes:54]: My understanding is that we need to be able to handle both

Comment (geoffreywatson) [reply conf/app.routes:54]: yes both ref-no for DES and referenceNumber for IF. Eventually we will remove ref-no (once IF fully up and running)

**Comment by geoffreywatson** [comment]
I think it's a good idea to create a IFStubController and copy over the 3 API's that we know will call IF (API 1495, 1167 & 1168)

**Comment by barrysims** [comment]
> I think it's a good idea to create a IFStubController and copy over the 3 API's that we know will call IF (API 1495, 1167 & 1168)

will do

**Pull request closed without merging**